xdsl/dialects/stencil.py

```diff
@@ -561,17 +561,20 @@ def parse_operand():
     def get(
         args: Sequence[SSAValue] | Sequence[Operation],
         body: Block | Region,
-        result_types: Sequence[TempType[Attribute]],
+        result_types: Sequence[TempType[Attribute]] | None = None,
+        bounds: StencilBoundsAttr | None = None,
     ):
-        assert len(result_types) > 0
-
+        assert result_types or bounds
```

Comment:
What happens if result_types is an empty sequence? Surely you want to have no results?
```suggestion
        assert result_types is not None or bounds is not None
```

Reply:
Yeah that's semantic!
Basically the op needs a non-empty result list or bounds, so the `(result_types = [], bounds = None)` case is invalid too :slightly_smiling_face:

Reply:
Would this work? `result_types: Sequence[TempType[Attribute]] = ()`

Reply:
Sure, but I'd still want to check it for non-emptiness? I'm confused

Reply:
This avoids the `None`/ empty ambiguity

Reply:
Done!